### PR TITLE
[FEATURE] Add and remove

### PR DESCRIPTION
**Pull request by Andyveloper**
# Requirements
- [x]  Remove all hardcoded items from the tasks array.
- [x] Create a new JavaScript file for the new functionality.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x]  Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for editing task descriptions.
- [x] By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- [x] All changes to the To Do List should be saved in local storage.

## Take into account
- All requirements completed

### CODE REVIEW

File: src/modules/newTask.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+export const saveEdit = () => {
+  const list = document.querySelector('.list-here');
+  const listItems = list.querySelectorAll('.list-items');
+  listItems.forEach((listItem) => {
+    const span = listItem.querySelector('span');
+    span.addEventListener('keydown', () => {
+      setData();
+    });
+  });
+};
```
Comment (roshan-bajgain) [inline src/modules/newTask.js]: - [ ] When I edit the task it edits properly but when I refresh it disappears and the edit task is not saved in local storage as well. Kindly Implement that function properly.

File: src/modules/localStorage.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { toDo } from '../index.js';
+import { createListElement } from './newTask.js';
+
+export function setData() {
+  localStorage.setItem('toDo', JSON.stringify(toDo));
+}
+
+export function getData() {
+  const savedData = JSON.parse(localStorage.getItem('toDo'));
+  return savedData;
```
Comment (roshan-bajgain) [inline src/modules/localStorage.js]: - [ ] Your local storage is not working properly, sometimes when I delete the second item it deletes all the items and when I add 3 items it saves only 2 items on local storage. Kindly fix it.
### Error
his

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Zeraltz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️

- [ ] [Optional] When a user clicks the check box then it should show completed as `true` on local storage but it is not updating kindly fix it and clear all buttons should clear all completed task but it is not working. Kindly fix it as it was the previous project requirements.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/localStorage.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  localStorage.setItem('toDo', JSON.stringify(toDo));
+}
+
+export function getData() {
+  const savedData = JSON.parse(localStorage.getItem('toDo'));
+  if (savedData) {
+    return savedData;
+  }
+  return [];
+}
```
Comment (FrederickMih) [inline src/modules/localStorage.js:13]: - [ ] Great job @Zeraltz :+1: your localStorage is now working properly.

**Review by FrederickMih — APPROVED** [review]
Hi @Zeraltz , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**